ui/src/app/applications/components/utils.tsx

```diff
@@ -1029,6 +1029,13 @@ export function getAppDefaultSource(app?: appModels.Application) {
     return app.spec.sources && app.spec.sources.length > 0 ? app.spec.sources[0] : app.spec.source;
 }
 
+export function getAppSources(app?: appModels.Application): appModels.ApplicationSource[] {
+    if (!app) {
+        return null;
+    }
+    return app.spec.sources;
```

Comment:
Could return the list of sources or a list with one source based on `app.spec.source`. This way the code could remove a few if/else logic

Reply:
The intent is really to get `app.spec.sources`.  I simplified the logic, and ended up not requiring this function.